pandas/core/indexes/base.py

```diff
@@ -732,6 +733,12 @@ def astype(self, dtype, copy=True):
             from .category import CategoricalIndex
             return CategoricalIndex(self.values, name=self.name, dtype=dtype,
                                     copy=copy)
+        elif is_datetime64tz_dtype(dtype):
+            # avoid FutureWarning from DatetimeIndex constructor.
+            from pandas import DatetimeIndex
+            tz = pandas_dtype(dtype).tz
+            return (DatetimeIndex(np.asarray(self))
+                    .tz_localize("UTC").tz_convert(tz))
```

Comment:
FYI, the code down on L747 is looking pretty dead to me (on master as well, since `is_extension_array_dtype(dtype)` was True for datetimetz dtype). Haven't tested though.

Reply:
there are probably a number of cases where this can be changed, cc @jbrockmendel